fastlane/helper/tool_name_formatting_helper.rb

```diff
@@ -0,0 +1,38 @@
+module Fastlane
+  module Helper
+    class ToolNameFormattingHelper
+      attr_accessor :content, :path
+
+      def initialize(content:, path:)
+        @content = content
+        @path = path
+      end
+
+      def find_tool_name_formatting_errors
+        errors = []
+        Fastlane::TOOLS.each do |tool|
+          lines = content.split("\n")
+          lines.each_with_index do |line, index|
```

Comment:
Switching your double loop to this:
```suggestion
        File.readlines(@path, mode: 'rb:BOM|UTF-8').each_with_index do |line, index|
          Fastlane::TOOLS.each do |tool|
```
Then stop passing/storing `content` to the helper (nor reading the file's content at call site) should improve the memory footprint of the algorithm, and solve your encoding problem.